--- a/res/res_timing_pthread.c
+++ b/res/res_timing_pthread.c
@@ -24,6 +24,7 @@
  */
 
 /*** MODULEINFO
+	<defaultenabled>no</defaultenabled>
 	<support_level>extended</support_level>
  ***/
 
--- a/res/res_ael_share.c
+++ b/res/res_ael_share.c
@@ -26,6 +26,7 @@
  */
 
 /*** MODULEINFO
+	<defaultenabled>no</defaultenabled>
 	<support_level>extended</support_level>
  ***/
 
--- a/res/res_phoneprov.c
+++ b/res/res_phoneprov.c
@@ -27,6 +27,7 @@
  */
 
 /*** MODULEINFO
+	<defaultenabled>no</defaultenabled>
 	<support_level>extended</support_level>
  ***/
 
--- a/res/res_clialiases.c
+++ b/res/res_clialiases.c
@@ -27,6 +27,7 @@
  */
 
 /*** MODULEINFO
+	<defaultenabled>no</defaultenabled>
 	<support_level>core</support_level>
  ***/
 
