--- a/channels/chan_sip.c
+++ b/channels/chan_sip.c
@@ -5308,10 +5308,11 @@ static int create_addr(struct sip_pvt *d
 	/* Get the outbound proxy information */
 	ref_proxy(dialog, obproxy_get(dialog, NULL));
 
-	if (addr) {
+	/* If we have an outbound proxy, don't bother with DNS resolution at all */
+	if (addr && !dialog->outboundproxy) {
 		/* This address should be updated using dnsmgr */
 		ast_sockaddr_copy(&dialog->sa, addr);
-	} else {
+	} else if (!dialog->outboundproxy) {
 
 		/* Let's see if we can find the host in DNS. First try DNS SRV records,
 		   then hostname lookup */
